extensions/Push/lib/Connector/Phabricator.pm

```diff
@@ -56,81 +48,97 @@ sub send {
-    # If so, then this doesn't automatically mean we need to make req to Phab?
-    # ======================================================
-    $is_public = is_public($bug);
+    my $is_public = is_public($bug);
+
+    my $phab_sync_groups = Bugzilla->params->{phabricator_sync_groups};
+    ThrowUserError('invalid_phabricator_sync_groups') unless $phab_sync_groups;
 
-    # Check to see if any synced groups have changed
-    # =================================================
-    my $phab_sync_groups = Bugzilla->params->{phabricator_sync_groups}
-        || ThrowUserError('invalid_phabricator_sync_groups');
     my $sync_group_names = [ split('[,\s]+', $phab_sync_groups) ];
 
     my $bug_groups = $bug->groups_in;
     my $bug_group_names = [ map { $_->name } @$bug_groups ];
 
     my @set_groups = intersect($bug_group_names, $sync_group_names);
 
-    # "If bug privacy groups do not have any matching synchronized groups,
-    # then leave revision private and it will have be dealt with manually."
-    # ==
-    # Is this: https://trello.com/c/9edMlHrl/399-2-when-locking-down-a-revision-because-of-unknown-security-groups-i-want-to-see-a-helpful-message-in-the-revision-comments-and-t
-    # ????
-    # Should we return PUSH_RESULT_OK; ?
-    # ========================================
     if (!$is_public && !@set_groups) {
+        $logger->info('Edge case for bug privacy; making revisions private.');
+
+        my @revisions = _get_attachment_revisions($bug);
+        foreach my $revision (@revisions) {
+            make_revision_private($revision->{phid});
+        }
+
+        my $error_message = 'Bug was made private due to unknown private group.';
+
+        add_comment_to_revision($error_message);
+        $bug->add_comment($error_message);
+
         ThrowUserError('invalid_phabricator_sync_groups');
+
+        return PUSH_RESULT_OK;
     }
 
-    # For effeciency, we only need to take these actions if bug not public
-    # =========================================
+    my $policy_phid;
+    my $subscribers;
     if(!$is_public) {
-        my $policy_phid = create_private_revision_policy($bug, \@set_groups);
-        my $subscribers = get_bug_role_phids($bug);
+        $policy_phid = create_private_revision_policy($bug, \@set_groups);
+        $subscribers = get_bug_role_phids($bug);
     }
 
+    my @revisions = _get_attachment_revisions($bug);
+
+    foreach my $revision (@revisions) {
+        my $revision_phid = $revision->{phid};
+
+        if($is_public) {
+            $logger->info('Making revision $rev_phid public:');
+            $logger->info($revision_phid);
+            make_revision_public($revision_phid);
+        }
+        else {
+            $logger->info('Updating custom group policy:');
+            $logger->info($revision_phid);
+            edit_revision_policy($revision_phid, $policy_phid, $subscribers);
+        }
+    }
+
+    return PUSH_RESULT_OK;
+}
+
+sub _get_attachment_revisions() {
```

Comment:
hold off. i am adding it as part of https://github.com/mozilla-bteam/bmo/pull/166 as it was supposed to be there already but my original push left it out accidentally.